images/bb/cmd/bb/main.go

```diff
@@ -62,8 +62,7 @@ func (m *Main) Serve(ctx context.Context) {
 	var bb *blackboard.Blackboard
 
 	// close when ready
-	ready := make(chan struct{})
-	markReady := func() { close(ready) }
+	readyCtx, markReady := context.WithCancel(context.Background())
 	var markReadyOnce sync.Once
```

Comment:
you can also remove the `Once` since additional context cancel calls are a no-op